src/main/java/org/humancellatlas/ingest/export/DefaultExporter.java

```diff
@@ -79,13 +82,16 @@ public void exportBundles(SubmissionEnvelope envelope) {
 
     @Override
     public void updateBundles(SubmissionEnvelope submissionEnvelope) {
-        Collection<MetadataDocument> documentsToUpdate = new ArrayList<>();
-        documentsToUpdate.addAll(metadataCrudService.findBySubmission(submissionEnvelope, EntityType.PROJECT));
-        documentsToUpdate.addAll(metadataCrudService.findBySubmission(submissionEnvelope, EntityType.BIOMATERIAL));
-        documentsToUpdate.addAll(metadataCrudService.findBySubmission(submissionEnvelope, EntityType.PROTOCOL));
-        documentsToUpdate.addAll(metadataCrudService.findBySubmission(submissionEnvelope, EntityType.PROCESS));
-        documentsToUpdate.addAll(metadataCrudService.findBySubmission(submissionEnvelope, EntityType.FILE));
-
+    	// FIXME why do we need a whole collection of metadocument here?
```

Comment:
The implementation of finding the bundles where a metadata belongs to is very inefficient. We are planning to restructure the Ingest bundle manifest to fix this.